CONTRIBUTING.md

```diff
@@ -319,8 +322,84 @@ func virtualMachines(ctx context.Context, a *azurerm, resourceType string, tags
 	return resources, nil
 }
 ```
+5. Optional: If the resource data is required to list other resources, you should add caching, e.g.: listing virtual machine extension requires the name of the virtual machines.
+
+	1. To enable this you should first change the resource code at `azurerm/resources.go`.
```

Comment:
IDK if this big blob of code is a good thing to have here, if we change the code API we have to be aware of it, i would much rather describe a bit and give links to cod than having blobs of code here.